test/claimZap.js

```diff
@@ -0,0 +1,239 @@
+    console.log('ve3TokenRewardPoolBalanceOfUserA After: ' + formatEther(lpRewardOfUserAAfter.toString()) + '\n');
+    assert.equal(
+        toBN(ve3TokenRewardPoolBalanceOfUserAAfter).minus(ve3TokenRewardPoolBalanceOfUserABefore),
+        web3.utils.toWei('10'),
+    );
+
+    // increase time, check rewards
+    starttime = await time.latest();
+    await time.increase(21 * 86400); // 21 days, 1 day = 86400 s
+    await time.advanceBlock();
+
+    const lockRewardPerToken = await ve3TokenLockRewardPool.rewardPerToken();
+    const stakeLockRewardPerToken = await ve3dLocker.rewardPerToken(veAsset.address);
+    const userARewardInlockRewardPool = await ve3TokenLockRewardPool.earned(userA);
+
+    await booster.earmarkRewards(0);
+
+    let options = 1 + 4 + 8; // ClaimVetoken, ClaimVe3Token, ClaimLockedVeToken
+    await zap.claimRewards([lPRewardPool.address, ve3TokenRewardPool.address], [ve3TokenLockRewardPool.address], [], [], 0, 0, 0, options, {from: userA});
+
+    console.log('get veAsset locking rewards:', lockRewardPerToken.toString(), stakeLockRewardPerToken.toString());
+    console.log('user A rewards in ve3TokenLockRewardPool from locking', userARewardInlockRewardPool.toString());
+
+    const veAssetBalanceAfterRewardClaimed = await veAsset.balanceOf(userA);
+    const ve3TokenClaimed = await ve3Token.balanceOf(userA);
+
+    console.log(
+        'userA earned veAsset:',
+        toBN(veAssetBalanceAfterRewardClaimed).minus(veAssetTokenBalanceBefore).toString(),
+    );
+    console.log('userA earned ve3Token:', toBN(ve3TokenClaimed).minus(ve3TokenBalanceBefore).toString());
+
+    assert.isAbove(Number((toBN(veAssetBalanceAfterRewardClaimed).minus(veAssetTokenBalanceBefore))), 0);
+
+    assert.equal(Number(toBN(ve3TokenClaimed).minus(ve3TokenBalanceBefore)), 0);
+
+    //advance time again
+    await time.increase(20 * 86400); //20 days
+    await time.advanceBlock();
+    console.log('advance time...');
+    await time.latest().then((a) => console.log('current block time: ' + a));
+    await time.latestBlock().then((a) => console.log('current block: ' + a));
+
+    await booster.earmarkRewards(0);
+
+    const userARewardInlockRewardPool3 = await ve3TokenLockRewardPool.earned(userA);
+    const userARewardInlpRewardPool3 = await lPRewardPool.earned(userA);
+    const userARewardInRewardPool3 = await ve3TokenRewardPool.earned(userA);
+    console.log('user A new reward in LockRewardPool', userARewardInlockRewardPool3.toString());
+    console.log('user A new reward in lpRewardPool', userARewardInlpRewardPool3.toString());
+    assert.isAbove(Number(userARewardInlpRewardPool3), 0);
+    console.log('user A new reward in RewardPool', userARewardInRewardPool3.toString());
+    assert.isAbove(Number(userARewardInRewardPool3), 0);
+
+    await zap.claimRewards([lPRewardPool.address, ve3TokenRewardPool.address], [ve3TokenLockRewardPool.address], [], [],0, 0, 0, options, {from: userA});
```

Comment:
the test not cover all logic ,by options parameters

Reply:
If I remember correctly I was told that we would create more tests in the future as we need them, right? @jetbrain10